website/docs/docs/tutorials/primary-api.md

```diff
@@ -99,6 +101,21 @@ const model = {
 };
 ```
 
+### State type restrictions
+
+Using plain objects as state is recommended, as it has been battle tested &
+proven to be stable.
+
+If you however want to use complex types as part of your state, you need to
+consider the following:
+
+- To support `class` instances, you need to ensure that the `class` is
+  [compatible with `immer`](https://immerjs.github.io/immer/complex-objects/).
+- To use `Map` or `Set` within your state,
+  [you need to enable it](/docs/introduction/browser-support.html#supporting-map-or-set-within-your-state).
+
+⚠️ Using complex types within the state may lead to unknown side effects.
```

Comment:
I'm not sure about this one - it's just a "cover your ass"-statement 😁

Keep or remove it?

Reply:
I say keep it. This covers three known cases, but there may be other things out there that haven't surfaced yet (seems like there would be limited use-cases for someone to actually do this, but what about `WeakMap` and `WeakSet` - if the polyfill doesn't cover those...). CYA never hurts.